odxtools/diaglayer.py

```diff
@@ -125,7 +129,22 @@ def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
 
         self.diag_layer_raw._resolve_odxlinks(odxlinks)
 
-    def _finalize_init(self, odxlinks: OdxLinkDatabase) -> None:
+    def __deepcopy__(self, memo: Dict[int, Any]) -> Any:
```

Comment:
Where is the deep copy needed? I can't find it in the diff

Reply:
that's for the `mksomersaultmodifiedpdx.py` example.

Reply:
(because `deepcopy()` makes a copy of all attributes of an object, it is advisable not to copy the new `_database` attribute. In some sense, what we really want is a variant of `deepcopy()` that only copies the fields of dataclasses.)